src/main/java/ua/softserveinc/tc/dao/impl/BookingDaoImpl.java

```diff
@@ -177,4 +189,27 @@ public void cancellActiveAndPlannedBookingsInRoom(Room room) {
         return bookings;
 
     }
+
+    /*
+     * Set the state to Cancelled, sum and duration to 0 for all booking with given
+     * path and value of path
+     *
+     * @param path the given path
+     * @param bookingId the given value of path
+     * @return the number of entities deleted
+     */
+    private int cancelBookingsWherePathAndValue(String path, long value) {
```

Comment:
It is bad Idea. Because then will be complicated usage this method for service methods. So, the best choice is put Transactional on service method.